index.js

```diff
@@ -426,28 +426,38 @@ class EnvisalinkPlatform {
                 this.log.debug("partitionUpdate: Partition data - " + partition.name + ' to ' + partition.envisakitCurrentStatus);
                 const partitionService = partition.accessory.getService(Service.SecuritySystem);
                 if (partitionService) {
-                    if (partition.homekitLastTargetState != partition.ENVISA_TO_HOMEKIT_TARGET[data.mode])
+                    var targetState = partition.ENVISA_TO_HOMEKIT_CURRENT[data.mode];
+
+                    if (partition.homekitLastTargetState != targetState)
                         {
-                            partitionService.updateCharacteristic(Characteristic.SecuritySystemCurrentState,partition.ENVISA_TO_HOMEKIT_CURRENT[data.mode]);
+                            if (
+                                targetState === Characteristic.SecuritySystemTargetState.STAY_ARM &&
+                                partitionService.getCharacteristic(Characteristic.SecuritySystemTargetState).value === Characteristic.SecuritySystemTargetState.NIGHT_ARM
+                            ) {
+                                targetState = Characteristic.SecuritySystemTargetState.NIGHT_ARM;
+                            }
```

Comment:
Hi,

I haven't researched using keypad text update to update alarm panel state... interesting approach. Does the keypad text always indicate "NIGHT" for each panel update event? 

The changes that would be required wouldn't occur in the files you are suggesting, it would have to occur in main envisakit class and would be transparent to Homebridge.  The classes interacting with Homebridge wouldn't change.